package/MDAnalysis/topology/RDKitParser.py

```diff
@@ -0,0 +1,285 @@
+                
+
+        # make Topology attributes
+        attrs = []
+        n_atoms = len(ids)
+
+        # * Attributes always present *
+
+        # Atom attributes
+        for vals, Attr, dtype in (
+            (ids, Atomids, np.int32),
+            (elements, Elements, object),
+            (masses, Masses, np.float32),
+            (aromatics, Aromaticities, bool),
+        ):
+            attrs.append(Attr(np.array(vals, dtype=dtype)))
+
+        # Bonds
+        bonds = []
+        bond_types = []
+        bond_orders = []
+        for bond in mol.GetBonds():
+            bonds.append((bond.GetBeginAtomIdx(), bond.GetEndAtomIdx()))
+            bond_orders.append(bond.GetBondTypeAsDouble())
+            bond_types.append(str(bond.GetBondType()))
+        attrs.append(Bonds(bonds, types=bond_types, order=bond_orders))
+
+        # * Optional attributes *
+
+        # Atom name
+        if names:
+            attrs.append(Atomnames(np.array(names, dtype=object)))
+        else:
+            for atom in mol.GetAtoms():
+                name = "%s%d" % (atom.GetSymbol(), atom.GetIdx())
+                names.append(name)
+            attrs.append(Atomnames(np.array(names, dtype=object)))
+
+        # Atom type
+        if atomtypes:
+            attrs.append(Atomtypes(np.array(atomtypes, dtype=object)))
+        else:
+            atomtypes = guessers.guess_types(names)
+            attrs.append(Atomtypes(atomtypes, guessed=True))
+
+        # Partial charges
+        if charges:
+            attrs.append(Charges(np.array(charges, dtype=np.float32)))
+        else:
+            pass # no guesser yet
+        
+        # PDB only
+        for vals, Attr, dtype in (
+            (altlocs, AltLocs, object),
+            (chainids, ChainIDs, object),
+            (occupancies, Occupancies, np.float32),
+            (tempfactors, Tempfactors, np.float32),
+        ):
+            if vals:
+                attrs.append(Attr(np.array(vals, dtype=dtype)))
+
+        # Residue
+        if any(resnums) and not any(val is None for val in resnums):
+            resnums = np.array(resnums, dtype=np.int32)
+            resnames = np.array(resnames, dtype=object)
+            segids = np.array(segids, dtype=object)
+            residx, (resnums, resnames, segids) = change_squash(
+                (resnums, resnames, segids), 
+                (resnums, resnames, segids))
+            n_residues = len(resnums)
+            for vals, Attr, dtype in (
+                (resnums, Resids, np.int32),
+                (resnums.copy(), Resnums, np.int32),
+                (resnames, Resnames, object)
+            ):
+                attrs.append(Attr(np.array(vals, dtype=dtype)))
+        else:
+            attrs.append(Resids(np.array([1])))
+            attrs.append(Resnums(np.array([1])))
+            residx = None
+            n_residues = 1
+
+        # Segment
+        if any(segids) and not any(val is None for val in segids):
```

Comment:
@cbouy Was just checking codecov to double check that there were enough tests and it looks like this branch of the code isn't tested at all (might be looking at an old commit). Could you double check that this is the case & add a test if so? https://codecov.io/gh/MDAnalysis/mdanalysis/src/dc482401df0d5c5839b2b37343b49a778987c2b1/package/MDAnalysis/topology/RDKitParser.py